tests/meltano/cli/test_state.py

```diff
@@ -281,6 +281,14 @@ def test_clear(self, state_service, cli_runner, state_ids) -> None:
                 job_state = state_service.get_state(state_id)
                 assert (not job_state) or (not job_state.get("singer_state"))
 
+    def test_clear_all(self, state_service, cli_runner, state_ids) -> None:
+        with mock.patch("meltano.cli.state.StateService", return_value=state_service):
+            result = cli_runner.invoke(cli, ["state", "clear", "--force", "--all"])
+            assert_cli_runner(result)
+            for state_id in state_ids:
+                job_state = state_service.get_state(state_id)
+                assert (not job_state) or (not job_state.get("singer_state"))
```

Comment:
suggestion: could we use `StateService.list_state` to check the before (> 0) and after (0) state ID count?